### DISCUSSION TITLE
Query datasets by creation or modification date using the CKAN API

### DISCUSSION THREAD

**Discussion by MichaelHarmsTIB** [ckan/ckan#6479] [Q&A]
Hello,
how can fetch the package_search by createdDate or lastmodified-date?
Is there a proper way to fetch all packages (I mean not package_list)?

**Comment by amercader** [comment]
You can use the `metadata_modified` and `metadata_created` fields in your queries. The syntax is a bit weird:

Datasets modified from a date until now:
```
q=metadata_modified:[2021-07-01T00:00:0.000Z TO NOW]
```

Datasets created between two dates:
```
q=metadata_created:[2021-07-01T00:00:0.000Z TO 2021-09-01T00:00:0.000Z]
```

For instance:

https://demo.ckan.org/api/action/package_search?q=metadata_modified%3A[2021-07-01T00%3A00%3A00.000Z+TO+NOW]+&sort=score+desc%2C+metadata_modified+desc

To get all datasets metadata from a site using `package_search` you need to page the results using the `rows` and `start` parameters, and keep doing requests until you don't get any more results. Most sites allow `rows` (the number of datasets returned) to be 1000, so if the site has less datasets than that you might only need one request.

**Reply by MichaelHarmsTIB** [reply]
Thank you!

**Reply by wardi** [reply]
You can also use the ckanapi CLI to get some or all your datasets using `package_search`, it will paginate automatically for you, e.g:
```bash
ckanapi search datasets -r https://demo.ckan.org
```
